src/Gui/DlgSettingsWorkbenchesImp.cpp

```diff
@@ -0,0 +1,434 @@
+
+
+    // 4: Autoloaded checkBox.
+    autoloadCheckBox = new QCheckBox(this);
+    autoloadCheckBox->setText(tr("Auto-load"));
+    autoloadCheckBox->setToolTip(tr("If checked, %1 will be loaded automatically when FreeCAD starts up").arg(wbDisplayName));
+    autoloadCheckBox->setEnabled(enableCheckBox->isChecked());
+
+    if (wbName.toStdString() == dlg->_startupModule) { // Figure out whether to check and/or disable this checkBox:
+        autoloadCheckBox->setChecked(true);
+        autoloadCheckBox->setEnabled(false);
+        autoloadCheckBox->setToolTip(tr("This is the current startup module, and must be autoloaded. See Preferences/General/Autoload to change."));
+    }
+    else if (std::find(dlg->_backgroundAutoloadedModules.begin(), dlg->_backgroundAutoloadedModules.end(),
+        wbName.toStdString()) != dlg->_backgroundAutoloadedModules.end()) {
+        autoloadCheckBox->setChecked(true);
+    }
+
+    // 5: Load button/loaded indicator
+    loadLabel = new QLabel(tr("Loaded"));
+    loadLabel->setAlignment(Qt::AlignCenter);
+    loadLabel->setEnabled(enableCheckBox->isChecked());
+    loadButton = new QPushButton(tr("Load"));
+    loadButton->setToolTip(tr("To preserve resources, FreeCAD does not load workbenches until they are used. Loading them may provide access to additional preferences related to their functionality."));
+    loadButton->setEnabled(enableCheckBox->isChecked());
+    connect(loadButton, &QPushButton::clicked, this, [this]() { onLoadClicked(); });
+    if (WorkbenchManager::instance()->getWorkbench(wbName.toStdString())) {
+        loadButton->setVisible(false);
+    }
+    else {
+        loadLabel->setVisible(false);
+    }
+
+    auto layout = new QHBoxLayout(this);
+    layout->addWidget(enableCheckBox);
+    layout->addWidget(iconLabel);
+    layout->addWidget(textLabel);
+    layout->addWidget(loadButton);
+    layout->addWidget(loadLabel);
+    layout->addWidget(autoloadCheckBox);
+    layout->setAlignment(Qt::AlignLeft);
+    layout->setContentsMargins(10, 0, 0, 0);
+}
+
+wbListItem::~wbListItem()
+{
+}
+
+void wbListItem::onLoadClicked()
+{
+    // activate selected workbench
+    Workbench* originalActiveWB = WorkbenchManager::instance()->active();
+    Application::Instance->activateWorkbench(objectName().toStdString().c_str());
+    Application::Instance->activateWorkbench(originalActiveWB->name().c_str());
+
+    // replace load button with loaded indicator
+    loadButton->setVisible(false);
+    loadLabel->setVisible(true);
+}
+
+void wbListItem::onWbActivated(bool checked)
+{
+    // activate/deactivate the widgets
+    iconLabel->setEnabled(checked);
+    textLabel->setEnabled(checked);
+    loadLabel->setEnabled(checked);
+    loadButton->setEnabled(checked);
+    autoloadCheckBox->setEnabled(checked);
+    if (!checked) //disabling wb disable auto-load.
+        autoloadCheckBox->setChecked(false);
+
+    // Reset the start combo items.
+    dlg->setStartWorkbenchComboItems();
```

Comment:
Rather than breaking encapsulation here, you'd better have to emit a signal and declare `setStartWorkbenchComboItems` as a slot, so that you can connect them together.

Reply:
Yes, I removed the dlg pointer.